au/math_test.cc

```diff
@@ -248,6 +262,43 @@ TEST(fmod, HandlesIrrationalCommonUnit) {
     EXPECT_THAT(fmod(radians(1), degrees(57)), IsNear(degrees(0.2958), degrees(0.0001)));
 }
 
+TEST(remainder, SameAsStdRemainderForNumericTypes) {
+    EXPECT_EQ(remainder(3.5, 3), std::remainder(3.5, 3));
+    EXPECT_EQ(remainder(2.5, 3), std::remainder(2.5, 3));
+}
+
+TEST(remainder, ReturnsSameTypesAsStdRemainderForSameUnitInputs) {
+    const auto expect_consistent_with_std_remainder =
+        expect_consistent_with([](auto x, auto y) { return remainder(x, y); },
+                               [](auto x, auto y) { return std::remainder(x, y); });
+
+    expect_consistent_with_std_remainder(meters(4), meters(3));
+    expect_consistent_with_std_remainder(meters(4.f), meters(3.f));
+    expect_consistent_with_std_remainder(meters(4.), meters(3.));
+    expect_consistent_with_std_remainder(meters(4.l), meters(3.l));
+    expect_consistent_with_std_remainder(meters(4), meters(3.f));
+    expect_consistent_with_std_remainder(meters(4), meters(3.l));
+    expect_consistent_with_std_remainder(meters(4.), meters(3.l));
+}
+
+TEST(remainder, MixedUnitsSupportedWithCasting) {
+    constexpr auto a = meters(1);
+    constexpr auto b = centi(meters)(11);
+    constexpr auto expected_result = centi(meters)(1);
+
+    EXPECT_THAT(remainder(a, b), IsNear(expected_result, make_quantity<Nano<Meters>>(1)));
+}
+
+TEST(remainder, HandlesIrrationalCommonUnit) {
+    EXPECT_THAT(remainder(radians(1), degrees(57)), IsNear(degrees(+0.2958), degrees(0.0001)));
+    EXPECT_THAT(remainder(radians(1), degrees(58)), IsNear(degrees(-0.7042), degrees(0.0001)));
+}
+
+TEST(remainder, CenteredAroundZero) {
+    EXPECT_THAT(remainder(degrees(90), revolutions(1)), IsNear(degrees(90), degrees(1e-9)));
```

Comment:
Are the underlying types integers? Can this be exactly equals?

Reply:
The inputs are integers, but the outputs aren't.

I expect we _could_ do exactly-equals here, in _practice_, but because the underlying STL function is using floating point math, I don't feel confident that I could guarantee it.